### PR TITLE
Fix on document upload

### PR DESCRIPTION
**Pull request by Stoakes**
- Fix error on view to upload a document for an Etude 
    - Before : 404 on /Documents/Upload/Etude/{etude_name}
    - Now : upload of a document for an etude is working again
 - Improve UI for Etude and Member selection on Upload Doctype view (use of select2 + min-width where required)
 - Fix reference selection for Doctypes (Since #100 it was displaying Etude id instead of name)

### CODE REVIEW

File: src/Mgate/PubliBundle/Controller/DocumentController.php [hunk 1]
```diff
@@ -65,24 +66,18 @@ public function voirAction(Document $documentType)
 
     /**
      * @Security("has_role('ROLE_SUIVEUR')")
+     * @param Request $request
+     * @param Etude $etude
+     * @return bool|\Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
```
Comment (theofidry) [inline src/Mgate/PubliBundle/Controller/DocumentController.php]: import those classes to avoid to use the FQCN

Comment (theofidry) [reply src/Mgate/PubliBundle/Controller/DocumentController.php]: minor CS thing, I find more readable to add spaces where appropriate:

```php
/**
 * @Security("has_role('ROLE_SUIVER')")
 *
 * @param Request $request
 * @param Etude   $etude
 *
 * @return bool|RedirectReponse|Response
 */
```

Comment (theofidry) [reply src/Mgate/PubliBundle/Controller/DocumentController.php]: `RedirectResponse` is unnecessary as already included by  `Response` and unless I'm missing something `bool` is never returned so can be removed and a `Response` return type can be added

File: src/Mgate/PubliBundle/Controller/DocumentController.php [hunk 2]
```diff
@@ -65,24 +66,18 @@ public function voirAction(Document $documentType)
...
         }
 
-        $options['etude'] = $etude;
-
-        if (!$response = $this->upload($request, false, $options)) {
-            // Si tout est ok
+        if (!$response = $this->upload($request, false, ['etude' => $etude])) {
+            $this->addFlash('success','Document mis en ligne');
             return $this->redirect($this->generateUrl('MgateSuivi_etude_voir', array('nom' => $etude->getNom())));
         } else {
```
Comment (theofidry) [inline src/Mgate/PubliBundle/Controller/DocumentController.php]: unnecessary `else` as you have an early return

File: src/Mgate/PubliBundle/Controller/DocumentController.php [hunk 3]
```diff
@@ -65,24 +66,18 @@ public function voirAction(Document $documentType)
...
             throw new AccessDeniedException('Cette étude est confidentielle !');
         }
 
-        $options['etude'] = $etude;
-
-        if (!$response = $this->upload($request, false, $options)) {
-            // Si tout est ok
+        if (!$response = $this->upload($request, false, ['etude' => $etude])) {
+            $this->addFlash('success','Document mis en ligne');
             return $this->redirect($this->generateUrl('MgateSuivi_etude_voir', array('nom' => $etude->getNom())));
```
Comment (theofidry) [inline src/Mgate/PubliBundle/Controller/DocumentController.php]: `array()` => `[]`

File: src/Mgate/PubliBundle/Form/Type/RelatedDocumentType.php [hunk 4]
```diff
@@ -22,39 +22,43 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-                    'class' => 'Mgate\\PersonneBundle\\Entity\\Membre',
-                    'choice_label' => 'personne.prenomNom',
-                    'required' => false,
-                    'configs' => array('placeholder' => 'Sélectionnez un étudiant', 'allowClear' => true), ))
-        ;
+                'label' => 'Document lié à l\'étudiant (optionnel)',
+                'class' => 'Mgate\\PersonneBundle\\Entity\\Membre',
+                'choice_label' => 'personne.prenomNom',
+                'required' => false,
+                'attr' => array('style' => 'min-width: 300px'),
```
Comment (theofidry) [inline src/Mgate/PubliBundle/Form/Type/RelatedDocumentType.php:60]: CSS should be added in the HTML, not the form builder... may be too big of a change though

File: src/Mgate/PubliBundle/Controller/DocumentController.php (hunk 1 above)
Comment (Stoakes) [reply src/Mgate/PubliBundle/Controller/DocumentController.php]: Hey :)

Thank you for reviewing that PR and these advices. 

Done in 07f617a8944128bbbfd042d5adc50a9b499c4e66

File: src/Mgate/PubliBundle/Controller/DocumentController.php (hunk 3 above)
Comment (Stoakes) [reply src/Mgate/PubliBundle/Controller/DocumentController.php]: Yes. However, there are plenty of `array()` calls. 
It is a non sense to fix only these ones. I will apply php-cs-fixer in a few days to fix all that.

File: src/Mgate/PubliBundle/Controller/DocumentController.php (hunk 2 above)
Comment (Stoakes) [reply src/Mgate/PubliBundle/Controller/DocumentController.php]: Thank you. Fixed

File: src/Mgate/PubliBundle/Form/Type/RelatedDocumentType.php (hunk 4 above)
Comment (Stoakes) [reply src/Mgate/PubliBundle/Form/Type/RelatedDocumentType.php:60]: I agree with that rules and try to apply it in most of the cases.  (`DocType/upload.html.twig` in that PR for instance)

In this situation, we have just one template `Document/upload.html.twig` which is used by several document upload forms. 
Thus template has no idea about the form structure and renders it programatically (just iterate on form fields). 
And apart from checking form name in the template (heavy and not very adaptative), I don't see any other solution to display fields with the appropriate styling.

**Review by theofidry — APPROVED** [review]

**Pull request merged**